src/GovUk.Education.ExploreEducationStatistics.Public.Data.Processor.Tests/Functions/ProcessNextDataSetVersionFunctionTests.cs

```diff
@@ -0,0 +1,278 @@
+                .Returns(NullLoggerFactory.Instance);
+
+            mock
+                .SetupGet(context => context.InstanceId)
+                .Returns(instanceId?.ToString() ?? Guid.NewGuid().ToString());
+
+            mock
+                .SetupGet(context => context.IsReplaying)
+                .Returns(isReplaying);
+
+            return mock;
+        }
+    }
+
+    public class CreateMappingsProcessingTests(
+        ProcessorFunctionsIntegrationTestFixture fixture)
+        : ProcessInitialDataSetVersionFunctionTests(fixture)
+    {
+        private const DataSetVersionImportStage Stage = DataSetVersionImportStage.ImportingMetadata;
+
+        [Fact]
+        public async Task Success()
+        {
+            var (initialDataSetVersion, _) = await CreateDataSet(DataSetVersionImportStage.Completing);
+
+            // TODO EES-4945 - amend metadata below for the initial DataSetVersion to partially match
+            // metadata from ProcessorTestData.AbsenceSchool.
+
+            var initialLocationMeta = DataFixture
+                .DefaultLocationMeta()
+                .WithDataSetVersionId(initialDataSetVersion.Id)
+                .ForIndex(0, s => s
+                    .SetLevel(GeographicLevel.LocalAuthority)
+                    .SetOptions(DataFixture
+                        .DefaultLocationLocalAuthorityOptionMeta()
+                        .GenerateList(2)
+                        .Select(meta => meta as LocationOptionMeta)
+                        .ToList()))
+                .ForIndex(1, s => s
+                    .SetLevel(GeographicLevel.RscRegion)
+                    .SetOptions(DataFixture
+                        .DefaultLocationRscRegionOptionMeta()
+                        .GenerateList(2)
+                        .Select(meta => meta as LocationOptionMeta)
+                        .ToList()))
+                .GenerateList();
+
+            var initialFilterMeta = DataFixture
+                .DefaultFilterMeta()
+                .WithDataSetVersionId(initialDataSetVersion.Id)
+                .WithOptions(() => DataFixture
+                    .DefaultFilterOptionMeta()
+                    .GenerateList(2))
+                .GenerateList(2);
+
+            var initialIndicatorMeta = DataFixture
+                .DefaultIndicatorMeta()
+                .WithDataSetVersionId(initialDataSetVersion.Id)
+                .GenerateList(2);
+
+            var initialTimePeriodMeta = DataFixture
+                .DefaultTimePeriodMeta()
+                .WithDataSetVersionId(initialDataSetVersion.Id)
+                .GenerateList(4);
+
+            await AddTestData<PublicDataDbContext>(context =>
+            {
+                context.LocationMetas.AddRange(initialLocationMeta);
+                context.FilterMetas.AddRange(initialFilterMeta);
+                context.IndicatorMetas.AddRange(initialIndicatorMeta);
+                context.TimePeriodMetas.AddRange(initialTimePeriodMeta);
+            });
```

Comment:
Don't think you need any of this now since you've changed it to do the work in-memory

Reply:
We're only reading the *next* data set version's meta in-memory, but the original data set version's meta is still read from the db as it's already in there and ready to go!

Reply:
Oh sorry

Do we do anything with it though? The test seems to pass if you remove all this

Reply:
Oh yeah of course, because this PR doesn't contain the "create mappings" work!  Cool, removed here, but it's reinstated in the 4945 PR!  Ta